src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -295,7 +301,62 @@ public static Complex Exp(Complex value)
         [SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly", MessageId = "Sqrt", Justification = "Sqrt is the name of a mathematical function.")]
         public static Complex Sqrt(Complex value)
         {
-            return FromPolarCoordinates(Math.Sqrt(value.Magnitude), value.Phase / 2.0);
+
+            if (value._imaginary == 0.0)
+            {
+                // Handle the trivial case quickly.
+                if (value._real < 0.0)
+                {
+                    return new Complex(0.0, Math.Sqrt(-value._real));
+                }
+                else
+                {
+                    return new Complex(Math.Sqrt(value._real), 0.0);
+                }
+            }
+            else
+            {
+
+                // One way to compute Sqrt(z) is just to call Pow(z, 0.5), which coverts to polar coordinates
+                // (sqrt + atan), halves the phase, and reconverts to cartesian coordinates (cos + sin).
+                // Not only is this more expensive than necessary, it also fails to preserve certain expected
+                // symmetries, such as that the square root of a pure negative is a pure imaginary, and that the
+                // square root of a pure imaginary has exactly equal real and imaginary parts. This all goes
+                // back to the fact that Math.PI is not stored with infinite precision, so taking half of Math.PI
```

Comment:
> so taking half of Math.PI does not land us on an argument with sine exactly equal to zero

Is this worded correctly?The sine of (pi / 2) should be 1, not 0; or am I misunderstanding what this says?